js/urlsearchparams.ts

```diff
@@ -0,0 +1,174 @@
+export class URLSearchParams {
+
+	private params: [string, string][] = [];
+
+	public constructor(init: string | Iterable<[string, string]> | Record<string, string> = '') {
+		if (typeof init === "string") {
+			// Overload: USVString
+			// If init is a string and starts with U+003F (?),
+			// remove the first code point from init.
+			if (init.charCodeAt(0) === 0x003F) init = init.slice(1);
+
+			for (const pair of init.split("&")) {
+				// Empty params are ignored
+				if (pair.length === 0) continue;
+				const position = pair.indexOf("=");
+				const name = pair.slice(0, position === -1 ? pair.length : position);
+				const value = pair.slice(name.length + 1);
+				this.append(decodeURIComponent(name), decodeURIComponent(value));
+			}
+			// Is object
+		} else if (Object(init) === init) {
+			if (Symbol.iterator in init) {
+				// Overload: sequence<sequence<USVString>>
+				for (const tuple of init[Symbol.iterator]()) {
+					this.append(tuple[0], tuple[1]);
+				}
+			} else {
+				// Overload: record<USVString, USVString>
+				for (const key of Object.keys(init)) {
+					this.append(key, init[key]);
+				}
+			}
+		}
+	}
+
+	/** Appends a specified key/value pair as a new search parameter.
+	 *
+	 *       searchParams.append('name', 'first');
+	 *       searchParams.append('name', 'second');
+	 */
+	public append(name: string, value: string): void {
```

Comment:
`public` is unnecessary

Reply:
Fixed this and all other occurrences in the next commit, thanks you 👍